dev/com.ibm.ws.microprofile.health/src/com/ibm/ws/microprofile/health/internal/AppTrackerImpl.java

```diff
@@ -262,7 +341,13 @@ public void applicationStopped(ApplicationInfo appInfo) {
         // Remove the stopped application from the appState map
         lock.writeLock().lock();
         try {
-            appStateMap.remove(appName);
+            if (appStateMap.containsKey(appName)) {
+               if (appStateMap.get(appName) != ApplicationState.STARTED) {
+                   appStateMap.replace(appName, ApplicationState.INSTALLED);
```

Comment:
How do we know if the app is completely removed from the dropins directory or just stopped?
From the code, it looks like if the state is not STARTED, we are assuming that the app is just stopped and not completed removed from the dropins directory? However, the state in the appStateMap never gets modified, when the application is stopping, in other words, the state will always be STARTED and never go into the first if statement logic, and always go into the else statement logic, no matter what.

Reply:
Reverted the change to original implementation